packages/amplify-codegen/src/commands/models.js

```diff
@@ -44,6 +45,85 @@ const readNumericFeatureFlag = key => {
   }
 };
 
+/**
+ * Returns the version number of package installed
+ * @param {string} target language target
+ * @returns
+ */
+const readDependencyVersionInstalled = (dependency, target) => {
+  // TODO: other platforms and error handling
+  const projectPath = pathManager.findProjectRoot();
+  switch (target) {
+    case 'java':
+      return '';
+    case 'swift':
+      return '';
+    case 'dart':
+      return '';
+    case 'javascript':
+      // there are a few ways to get the version of an installed package
+      // just choosing one for PoC
+      // Will assess the best option later
+      const dependencyInstallPath = path.join(projectPath, 'node_modules', dependency, 'package.json');
+      if (fs.pathExistsSync(dependencyInstallPath)) {
```

Comment:
I think we need to check for the lock file rather than the package.json file itself.